src/pages/DatasetRegistration.js

```diff
@@ -101,6 +101,59 @@ class DatasetRegistration extends Component {
       prev.dacList = dacs;
       return prev;
     });
+    if (!fp.isEmpty(this.state.updateDataset)) {
+      this.prefillDatasetFields(this.state.updateDataset);
+    }
+  };
+
+  async init() {
+    const { datasetId } = this.props.match.params;
+    let updateDataset = {};
+    // update dataset case
+    if (!fp.isNil(datasetId)) {
+      DataSet.getDataSetsByDatasetId(datasetId).then(data => {
+        updateDataset = data;
+        // redirect to blank form if dataset id is invalid or inaccessible
+        if (fp.isEmpty(updateDataset) || fp.isNil(updateDataset.dataSetId)) {
+          this.props.history.push('/dataset_registration');
+          ReactTooltip.rebuild();
+        }
+        else {
+          this.setState(prev => {
+            prev.updateDataset = updateDataset;
+          });
+        }
+      });
+    }
+  };
+
+  // fill out the form fields with old dataset properties if they already exist
+  prefillDatasetFields(dataset) {
+    let name = dataset.properties.find(p => p.propertyName === "Dataset Name");
```

Comment:
I did something [very similar here](https://github.com/DataBiosphere/duos-ui/pull/689/files#diff-25f4b410d86206f6d12fcd985e2ffdc8b261c2973e139fcd9e02ca4a9a79ff4cR51)
It's a little less code and a little more expressive to do something like this:  
```suggestion
    let name = fp.find({propertyName: "Dataset Name"})(dataset.properties);
```